### PR TITLE
Add more error handling for facility user merging and facility deletion

### LINKED ISSUE #12486: On-my-own learner who joins a facility cannot open class assignements
## Observed behavior
Issue was first observed on a RPi server with Motorola phone, and replicated on Ubuntu 24.04 server with Xiaomi phone.

- Android device was set up as `On my own` 
- Learner account was then _transferred_ to the server using the *Change facility* workflow
- On the server facility the Learner account was enrolled into an already existing class with 1 quiz and 1 lesson
- On the device's Home screen, *Learner* is able to see the class they have been assigned to, and both assignments, but each time they try to open them, they are redirected to the Library page instead.
- Syncing has completed successfully, and  *Learner* can open and view each of the resources that are in the quiz and lesson, they just cannot view them _within_ the quiz and/or lesson.

https://github.com/user-attachments/assets/e52a3fb7-478f-4057-8635-1a8fc9281ae1

## Errors and logs
[Xiaomi-Android-13-logcat.txt](https://github.com/user-attachments/files/16375247/Xiaomi-Android-13-logcat.txt)

[db-and-logs.zip](https://github.com/user-attachments/files/16375578/db-and-logs.zip)

## Expected behavior
Learner who joins a facility and is enrolled in a class must be able to see and open the class assignments.

## User-facing consequences
Inability to open and complete the class assignments after changing the facility.

## Steps to reproduce
…

## Context

 * Kolibri version: 0.17b3
 * Operating system: RPi and Ubuntu 24.04 servers, Motorola G51 (Android 12) and Xiaomi 10C (Android 13) LoDs

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Moves IsSelf permissions check into tasks.py as it is specific to the arguments for the mergeuser task
* Updates it to check _either_ the local user id or the remote user id, so after merge, the user is still able to check the task.
* Handles an edge case where we were trying to check if the user was an admin for a facility that no longer existed
* Tries to prevent more weirdness by adding an explicit signal to clear the request.user cache when a user is deleted
* Calls this signal explicitly after calling `delete_facility` in the mergeuser task because the `delete_facility` function disables post_delete signals.
* Updates token generation for merge user flow to use only the user_id, so we can set this when first creating the task
* Updated handling for 403s to assume success, as it generally indicates that the user has been deleted and thus successfully merged
* Opportunistically update the global learn `inClasses` boolean when we fetch class data from the backend.

## References
Attempts to fix #12486 

## Reviewer guidance
While this only seems to happen on Android, I think it's more to do with unhandled errors arising from the process taking longer on Android, so hitting certain coincidences on Android that we might not see elsewhere.

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/user_profile/assets/src/views/ChangeFacility/MergeFacility.vue [hunk 1]
```diff
@@ -228,16 +228,34 @@
...
+              if (startedTask.status == TaskStatuses.COMPLETED) {
+                isPolling = false;
+              } else if (startedTask.status === TaskStatuses.FAILED) {
+                TaskResource.clear(taskId.value); // start a new one
+                isTaskRequested = false;
+                taskError.value = true;
+              }
+            })
+            .catch(err => {
+              if (err.status === 403 && task.value) {
```
Comment (rtibbles) [inline kolibri/plugins/user_profile/assets/src/views/ChangeFacility/MergeFacility.vue]: Should be `error.response.status`

File: kolibri/plugins/user_profile/assets/src/views/ChangeFacility/MergeFacility.vue [hunk 2]
```diff
@@ -228,16 +228,34 @@
...
+                // If we get a 403, it means that our currently logged in user
+                // does not have permission to access the task. This can happen
+                // if the user has been deleted by the task as intended, so assume
+                // the task has completed successfully.
+                isPolling = false;
```
Comment (rtibbles) [inline kolibri/plugins/user_profile/assets/src/views/ChangeFacility/MergeFacility.vue]: Don't need to set this twice.

**Review by radinamatic — APPROVED** [review]
Blazingly fast changing facility, learner can see all the class assignments! 👏🏽 🙇🏽‍♀️ :shipit:

**Review by marcellamaki — APPROVED** [review]
Did live code review walk through with Richard. Notes were included as his comments typed inline above. Good to go

**Pull request merged**